### PR TITLE
Improve install tests (setting `newArchEnabled` and `hermesEnabled` correctly on Android)

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

We had a bug in the way the install tests configured the Android test to run with Hermes disabled.
Also, we now test throwing in code called from C++, because that has historically been (and is currently) broken (because of https://github.com/facebook/react-native/issues/36052).

This also makes it so PRs updating the install test workflow won't post and spam the team channel anymore.

### CODE REVIEW

File: install-tests/react-native/cli.ts [hunk 1]
```diff
@@ -182,18 +182,20 @@ yargs(hideBin(process.argv))
...
+      const localPropertiesContent =
+        "\n# Install test overwrites below\n\n" +
+        Object.entries(localProperties)
+          .map(([k, v]) => `${k}=${v}`)
+          .join("\n");
 
-      console.log(`Writing local.properties to ${localPropertiesPath}`);
-      fs.writeFileSync(localPropertiesPath, localPropertiesContent);
+      console.log(`Appending gradle properties to ${localPropertiesPath}`);
+      fs.appendFileSync(localPropertiesPath, localPropertiesContent);
```
Comment (kraenhansen) [inline install-tests/react-native/cli.ts:198]: This fixes configuring hermes and new architecture.

File: .github/workflows/install-test-react-native.yml [hunk 2]
```diff
@@ -18,6 +18,7 @@ defaults:
 
 jobs:
   install:
+    name: Install Test on React Native ${{ matrix.platform == 'ios' && 'iOS' || 'Android' }} realm@${{ matrix.realm-version }}, react-native@${{ matrix.react-native-version }} new architecture ${{ matrix.new-architecture && 'enabled' || 'disabled' }} running ${{ matrix.engine }}
```
Comment (kraenhansen) [inline .github/workflows/install-test-react-native.yml:21]: Making the name of the job a bit more readable:

File: install-tests/react-native/patches/App.js [hunk 3]
```diff
@@ -33,13 +37,29 @@ const App = () => {
...
+
     // Read the persons out of the database again
     const message =
       "Persons are " +
       realm
         .objects("Person")
         .map((p) => p.name)
-        .join(", ");
+        .join(", ") +
+      suffix;
```
Comment (takameyer) [inline install-tests/react-native/patches/App.js:62]: If i'm understanding this right, this will output a "!" when successful?  Is that meaningful?

Comment (kraenhansen) [reply install-tests/react-native/patches/App.js:62]: > Is that meaningful?

Not really 🙂 I just need some string to round-trip and expect.

**Pull request merged**